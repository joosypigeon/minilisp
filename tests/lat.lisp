(define (null? x)
    (if (equal? x '())
        #t
        #f))

(define (lat l)
    (cond
        ((null? l) #t)
        ((atom? l) #t)
        ((atom? (car l)) (lat (cdr l)))
        (#t #f)))



(print (lat '(1 2 3))) ; => #t
(print (lat '(1 2 (3 4)))) ; => #f
(print (lat '((1 2) (3 4)))) ; => #f
(print (lat '((1 2) (3 4) (5 6)))) ; => #f
(print (lat '"hello")) ; => #t
(print (lat '())) ; => #t
(print (lat 'a)) ; => #t