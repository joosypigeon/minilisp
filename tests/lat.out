#t
NIL
NIL
NIL
#t
#t
#t
